commands/proxy/proxy-set.ts

```diff
@@ -22,22 +23,30 @@ export class ProxySetCommand extends ProxyCommandBase {
 	}
 
 	public async execute(args: string[]): Promise<void> {
-		let hostname = args[0],
-			port = args[1],
-			username = args[2],
-			password = args[3];
+		let urlString = args[0],
+			username = args[1],
+			password = args[2];
+
+		const noUrl = !urlString;
+		if (noUrl) {
+			if (!isInteractive()) {
+				this.$errors.fail("Console is not interactive - you need to supply all command parameters.");
+			} else {
+				urlString = await this.$prompter.getString("Url", { allowEmpty: false });
+			}
+		}
 
-		const noHostName = !hostname;
+		const urlObj = parse(urlString);
+		let port = urlObj.port && +urlObj.port || HttpProtocolToPort[urlObj.protocol];
```

Comment:
Theoretically there is a way for `port` to be undefined if `HttpProtocolToPort[urlObj.protocol]` returns `undefined`. This would happen in the unlikely case where the user enters an url address which makes use of neither `http` nor `https` protocol.